docs/tutorials/qsimcirq_gcp.md

```diff
@@ -0,0 +1,357 @@
+
+Select `Connect to local runtime`. You will see the UI:
+
+<img src="../images/colab_settings.png" width="500"/>
+
+This is where you past the edited URL from the previous section. Then click `Connect`
+
+<img src="../images/colab_success.png" width="300"/>
+
+You should now be able to run the cells in the notebook.
+
+### Large circuit
+A large circuit is provided with 32 qubits and a depth of 14 gate operations.  Run it here:
+
+
+<a target="_blank" href="https://colab.research.google.com/github/quantumlib/qsim/blob/master/docs/tutorials/qsimcirq.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Run q32d14.ipynb.</a>
+
+This large circuit will not succeed if you attempt to run the default runtime. Ensure that you repeat the connection procedure to the local runtime.
+
+
+## Optional: Connect to Jupyter directly
+
+Once you have port 8888 forwarded and the container running on your GCE VM, it is easy to connect directly to Jupyter without using Colab.
+
+In the previous step, you copied a URL like below. It is easy to just copy that URL and paste it directly into a browser running on your local machine.
+
+```less
+ http://127.0.0.1:8888/?token=7191178ae9aa4ebe1698b07bb67dea1d289cfd0e0b960373
+```
+
+
+ In the browser you should now see the Jupyter UI:
+
+
+![alt_text](../images/image2.png )
+
+
+To see the notebooks, navigate to  qsim > docs > tutorials. You will see:
+
+![alt_text](../images/image12.png )
+
+
+Click on qsimcirq.ipynb. This will load the notebook.
+
+You can now run these cells as you would in any notebook.
+
+![alt_text](../images/image1.png )
+
+If you choose to modify the notebook, you can save it on the *qsim-1* VM from *File* -> *Save As*, and saving to `/homedir/mynotebook.ipynb`.  This will save in your home directory on your VM. If you intend to destroy the VM after this tutorial, either download the notebooks from the VM or save directly from your browser.
+
+![alt_text](../images/image4.png )
+
+## Run interactively
+
+To run interactively within the container, you can open a second shell window to the VM as you did above, but without the port forwarding:
+
+```bash
+gcloud compute ssh [YOUR_VM_NAME]
+```
+
+Now, find the container ID with docker ps:
+
+```
+$ docker ps
+```
+You should see output something like:
+
+```less
+CONTAINER ID        IMAGE                                       COMMAND                  CREATED             STATUS              PORTS                    NAMES
+8ab217d640a3        gcr.io/quantum-291919/jupyter_qsim:latest   "jupyter-notebook --…"   2 hours ago         Up 2 hours          0.0.0.0:8888->8888/tcp   dazzling_lovelace
+```
+
+
+The `CONTAINER_ID` is a UID something like "8ab217d640a3". Now you can connect to the container:
+```console
+    $ docker exec -it [CONTAINER_ID] /bin/bash
+
+### Build the circuit
+
+To import the libraries and build the circuit, copy and paste following into the VM window.
+
+
+```python
```

Comment:
Copy-pasting this as a block misbehaves due to unexpected indent. Same for other python blocks below.

Reply:
@lamberta 

Is there a better way to represent a code block that allows proper cut and paste?

Reply:
Hello. Yes, take a look at this [docker.md](https://github.com/tensorflow/docs/blob/master/site/en/install/docker.md) doc (on [website](https://www.tensorflow.org/install/docker)).

Can use something like:

```
<pre class="devsite-terminal devsite-click-to-copy">
docker run [-it] [--rm] [-p <em>hostPort</em>:<em>containerPort</em>] tensorflow/tensorflow[:<em>tag</em>] [<em>command</em>]
</pre>
```

Reply:
Is it possible to just remove the leading spaces in these lines, as with the "less" block below?